### PR TITLE
[OAuth] Provider\SmashCast - initial, refs #51

### PR DESCRIPTION
**Pull request by ovr**
Hey!

Type: bug fix | new feature | code quality | documentation

Link to issue:

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/SocialConnect/auth/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:

Thanks :smiley_cat:

### CODE REVIEW

File: src/OAuth2/Provider/SmashCast.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+        return $result->user_name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentity(AccessTokenInterface $accessToken)
+    {
+        // @todo Find a problem with this code
```
Comment (ovr) [inline src/OAuth2/Provider/SmashCast.php]: ping @jens1o

Possible you can help me on my way, I cannot get it works with `auth/login` method on SmashCast API

File: src/OAuth2/Provider/SmashCast.php [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        /*$response = $this->httpClient->request(
+            $this->getBaseUri() . 'auth/login',
+            [
+                'app' => 'desktop', // @any app name, not working, I was using JSON and not working..
```
Comment (jens1o) [inline src/OAuth2/Provider/SmashCast.php]: Note: You can create these apps in the developers option in your account.

File: src/OAuth2/Provider/SmashCast.php [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                'authToken' => $accessToken->getToken()
+            ],
+            Client::POST,
+            [
+                'Content-Type' => 'application/x-www-form-urlencoded'
```
Comment (jens1o) [inline src/OAuth2/Provider/SmashCast.php]: You need to do a json header. So yes, it is a post request, but the body is in json, not a plain form.

File: src/OAuth2/Provider/SmashCast.php (hunk 2 above)
Comment (ovr) [inline src/OAuth2/Provider/SmashCast.php]: @jens1o Should i use `socialconnect`, OAuth app name? or something else?

image

Comment (ovr) [inline src/OAuth2/Provider/SmashCast.php]: I am getting every time 

```json
{"success":true,"error":false,"error_msg":"auth_failed"}
```

success: true, error: false but with `error_msg` lol

Comment (jens1o) [inline src/OAuth2/Provider/SmashCast.php]: Can you push your current code?

Comment (ovr) [inline src/OAuth2/Provider/SmashCast.php]: @jens1o I am using REST client for testing

image

Comment (jens1o) [inline src/OAuth2/Provider/SmashCast.php]: how do you get your access token?

Comment (jens1o) [inline src/OAuth2/Provider/SmashCast.php]: anyway, you may want to stop this approach. I'm fine with the current situation. :) I don't want to get you to get mad.

Comment (ovr) [inline src/OAuth2/Provider/SmashCast.php]: > how do you get your access token?

**Comment by ovr** [comment]
Implemented in `master` and release in `1.4.0` 🍾

**Pull request closed without merging**